### PR TITLE
INDY-1999: Transaction Endorser design

### PR DESCRIPTION
**Pull request by ashcherbakov**
Signed-off-by: ashcherbakov <alexander.sherbakov@dsr-corporation.com>

### CODE REVIEW

File: design/transaction_endorder.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+- If there is no `endorser` field, then it's assumed that the original author (`identifier`) is the endorser. 
+
+## libindy API to support endorsers
+```rust=
+/// Append Endorser to existing request expecting that the transaction will be sent by the specified Endorser.
+///
+///
+/// #Params
+/// request_json - original request
+/// text and endorser_did - DID of the Endorser that will submit the transaction.
```
Comment (jovfer) [inline design/transaction_endorder.md]:
```suggestion
/// endorser_did - DID of the Endorser that will submit the transaction. 
```

File: design/transaction_endorder.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+///
+///
+/// #Params
+/// request_json - original request
+/// text and endorser_did - DID of the Endorser that will submit the transaction. 
+///                         The Endorser's DID must be present on the ledger.
+/// cb: Callback that takes command result as parameter. 
+///     The command result is a request JSON with Endorser field appended.
+#[no_mangle]
+pub extern fn indy_append_endorser(command_handle: CommandHandle,
```
Comment (jovfer) [inline design/transaction_endorder.md]: Should we include "request" in the API call name as well?
like "indy_append_request_endorser"

Comment (ashcherbakov) [reply design/transaction_endorder.md]: Yes, let's do it.

**Comment by ashcherbakov** [comment]
(ci) test this please

**Pull request merged**